src/controller/CHIPDeviceController.h

```diff
@@ -367,6 +375,32 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public Rendezvous
      */
     CHIP_ERROR CloseBleConnection();
 #endif
+    /**
+     * @brief
+     *   Discover devices advertising as commissionable that match the long discriminator.
+     * @return CHIP_ERROR   The return status
+     */
+    CHIP_ERROR DiscoverCommissioningLongDiscriminator(uint16_t long_discriminator);
+
+    /**
+     * @brief
+     *   Discover all devices advertising as commissionable.
+     * @return CHIP_ERROR   The return status
+     */
+    CHIP_ERROR DiscoverAllCommissioning();
+
+    /**
+     * @brief
+     *   Helper function to print device info for device selection.
+     */
+    void PrintDiscoveredDevices();
+
+    /**
+     * @brief
+     *   Returns information about discovered devices.
+     * @return const CommissionableNodeData* info about the selected device. May be nullptr if no information has been returned yet.
+     */
+    const Mdns::CommissionableNodeData * GetDiscoveredDevice(int idx);
```

Comment:
This is likely going to be called from app code at some point, but is racy if called from a different thread than the main event loop. What is the way to ensure it is called in right context?

It also seems to be missing a getter for the number of devices discovered.

Reply:
it returns nullptr if the idx isn't filled, so you can just call until nullptr is returned. Per racing - is there a solution for multi-threaded access at this point?

Reply:
Well, it should be made clear in the comment (and from assertion about current thread) that this is called in main CHIP event loop thread.